src/create-element.js

```diff
@@ -100,3 +79,35 @@ export function Fragment(props) {
  */
 export const isValidElement = vnode =>
 	vnode != null && vnode.constructor === undefined;
+
+export function createBackingNode(vnode, original) {
+	const backingNode = {
+		key: vnode.props ? vnode.props.key : undefined,
+		ref: vnode.props ? vnode.props.ref : undefined,
+		type: vnode.type,
+		_node: vnode,
+		_children: null,
+		_parent: null,
+		_depth: 0,
+		_dom: null,
+		// _nextDom must be initialized to undefined b/c it will eventually
+		// be set to dom.nextSibling which can return `null` and it is important
+		// to be able to distinguish between an uninitialized _nextDom and
+		// a _nextDom that has been set to `null`
+		_nextDom: undefined,
+		_component: null,
+		_original: original
+	};
+
+	if (original == null) backingNode._original = vnode;
+
+	if (vnode.props && vnode.props.ref) {
+		delete vnode.props.ref;
```

Comment:
This could possibly be the slow part (together with 109)

Reply:
For performance, it's cheaper to create a new object than to mutate the shape of the current one. If we were to copy props to `backingNode.props` here, the copy could also exclude these keys:

```js
let props = {},
    originalProps = vnode.props,
    key,
    ref;

if (originalProps != null) {
    for (let i in originalProps) {
        const value = originalProps[i];
        if (i === 'key') key = value;
        else if (i === 'ref') ref = value;
        else props[i] = value;
    }
}

const backingNode = {
    props,
    key,
    ref,
    // ...
}
```

This way, the `originalProps[i]` access will be polymorphic, but nothing else. Elsewhere in the renderer, `backingNode.props` can be assumed to be an object, whereas currently we have to check if `vnode.props` is non-null. We might even be able to get rid of the current property copying loop in `h()`.